torchvision/datasets/imagenet.py

```diff
@@ -118,41 +115,95 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
-    val_wnids = [idx_to_wnid[idx] for idx in val_idcs]
-    return wnid_to_classes, val_wnids
-
+def parse_devkit_archive(archive, meta_file=None):
+    """
```

Comment:
I would create a dosctring since this function is now not only used internally. Same for `parse_{train|var}_archive()`